test/xbgpu/test_engine.py

```diff
@@ -638,8 +668,23 @@ async def _send_data(
 
                 corrprod_results[i][j] = ig_recv["xeng_raw"].value
 
+        # TODO: NGC-1172 The tweaks to process beam data below rely on
+        # `batch_indices` to be contiguous. The check below is temporary until
+        # the BPipeline is able to handle missing data.
+        assert sorted(batch_indices) == (
```

Comment:
Remove the `sorted`. You've got assumptions below that e.g. `batch_indices[0]` is the smallest index.